datafusion/optimizer/src/analyzer/expand_wildcard_rule.rs

```diff
@@ -301,4 +302,32 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn plan_having_wildcard_projection() -> Result<()> {
```

Comment:
We could test this kind of test in slt with `explain`. I prefer only adding rust test if the test is not trivial to stay in slt